mlir/lib/Dialect/Arith/IR/ArithOps.cpp

```diff
@@ -2314,7 +2314,8 @@ OpFoldResult arith::SelectOp::fold(FoldAdaptor adaptor) {
     return trueVal;
 
   // select %x, true, false => %x
-  if (getType().isInteger(1) && matchPattern(adaptor.getTrueValue(), m_One()) &&
+  if (getType().isSignlessInteger(1) &&
```

Comment:
AFAIK, `arith.select` does only allow signless types for the condition type. At least simply changing `%sel` from `i1` to `ui1` throws the following `error: use of value '%sel' expects different type than prior uses: 'i1' vs 'ui1'`.
But I agree, in case that this limitation will be lifted at some day, comparing the type directly would be better.